ra/registration-authority.go

```diff
@@ -381,17 +381,41 @@ func (ra *RegistrationAuthorityImpl) UpdateAuthorization(base core.Authorization
 }
 
 // RevokeCertificate terminates trust in the certificate provided.
-func (ra *RegistrationAuthorityImpl) RevokeCertificate(cert x509.Certificate) (err error) {
+func (ra *RegistrationAuthorityImpl) RevokeCertificate(cert x509.Certificate, reasonCode int, regID *int64) (err error) {
 	serialString := core.SerialToString(cert.SerialNumber)
-	err = ra.CA.RevokeCertificate(serialString, 0)
+	err = ra.CA.RevokeCertificate(serialString, reasonCode)
+
+	state := "Failure"
+	defer func() {
+		// AUDIT[ Revocation Requests ] 4e85d791-09c0-4ab3-a837-d3d67e945134
+		// Needed:
+		//   Serial
+		//   CN
+		//   Revocation reason
+		//   Error (if there was one)
+		revMsg := fmt.Sprintf(
+			"Revocation - State: %s, Serial: %s, CN: %s, DNS Names: %s, Reason: %s",
+			state,
+			serialString,
+			cert.Subject.CommonName,
+			cert.DNSNames,
+			core.RevocationReasons[reasonCode],
+		)
+		// Check regID is set, if not revocation came from the admin-revoker tool
```

Comment:
This is super confusing and unclear how a pointer being nil and coming from the admin-revoker is connected.

To make this a little more clear, how about a RevokeCertificateWithoutRegistration that only the admin-revoker uses? Both of them can just pass to the same command.

(What sucks is that we don't have an audit trail for who was running the admin-revoker when they did this or why.)